ios-base/DroidKaigi 2020/Announcements/ViewModels/AnnouncementsViewModel.swift

```diff
@@ -0,0 +1,54 @@
+import Foundation
+import ios_combined
+import RxCocoa
+import RxSwift
+
+protocol AnnouncementsViewModelType {
+    // Output
+    var announcements: Driver<[Announcement]> { get }
+    var error: Driver<KotlinError?> { get }
+
+    // Input
+    func viewDidLoad()
+    func pullToRefresh()
+}
+
+final class AnnouncementsViewModel: AnnouncementsViewModelType {
+    let announcements: Driver<[Announcement]>
+    let error: Driver<KotlinError?>
+
+    private let viewDidLoadReplay = PublishRelay<Void>()
+    private let pullToRefreshRelay = PublishRelay<Void>()
+
+    private let disposeBag = DisposeBag()
+
+    init(
+        provider: AnnouncementsDataProviderProtocol
+    ) {
+        let announcementsRelay = BehaviorRelay<[Announcement]>(value: [])
+        let errorRelay = BehaviorRelay<KotlinError?>(value: nil)
+
+        announcements = announcementsRelay.asDriver()
+        error = errorRelay.asDriver()
+
+        let fetchResult = Observable.merge(viewDidLoadReplay.asObservable(), pullToRefreshRelay.asObservable())
+            .flatMap { provider.fetch().asObservable().materialize() }
+            .share()
+
+        fetchResult.map { $0.element }
+            .filterNil()
```

Comment:
You might be able to use `compactMap`. ref: https://github.com/DroidKaigi/conference-app-2020/pull/588#discussion_r371886343

Reply:
I fixed.

https://github.com/DroidKaigi/conference-app-2020/pull/628/commits/608b1e0edaf61f61b9de80e680e1acf83daccd78